pkg/apiserver/filters/kubeapiserver.go

```diff
@@ -35,8 +35,8 @@ func WithKubeAPIServer(handler http.Handler, config *rest.Config, failed proxy.E
 
 			// Do not cover k8s client authorization header
 			req.Header.Del("Authorization")
-
 			httpProxy := proxy.NewUpgradeAwareHandler(&s, defaultTransport, true, false, failed)
+			httpProxy.UpgradeTransport = proxy.NewUpgradeRequestRoundTripper(defaultTransport, defaultTransport)
```

Comment:
What's this for? Please add more comments to clarify

Reply:
NewUpgradeRequestRoundTripper takes two round trippers - one for the underlying TCP connection, and one that is able to write headers to an HTTP request. 

see also: https://github.com/kubernetes/kubernetes/pull/88781

Reply:
Please add more comments, this will disable http2 for all requests. I think it's a temporary solution, we'd better switch back in future once upgraded to v1.18